### ISSUE TITLE
Document image sizes for components

### ISSUE THREAD

**Issue by stephaniehobson** [mozilla/protocol#201]
It'd be nice to have image dimensions documented so that assets can be delivered separately from components.

For example, someone could provide a file and say, "This is the image for the careers card" and we could look and see that card is a half column card at 3:2 ratio and know to create images at 1200x800 and 600x400.

**Comment by stephaniehobson** [comment]
@craigcook I think you had thoughts on this?

**Comment by craigcook** [comment]
Agreed. For any components with images we should make sure we include a minimum size requirement. We do have size recommendations in the notes for most components but a few might be missing them. We might also want to clarify that size recommendations are for low resolutions and we need most images delivered in two sizes.

This became an issue on the About page where we have medium (half-width) cards and the low-res images we were given were too small so they got upsized in wide windows. They were sized in Sketch for medium desktop screens and not for wide desktops (the Sketch file for that page maxes out at 1024). Really it would be ideal to get assets delivered separately, not have to extract them from a Sketch mockup. Sketch is not the web.

While I'm griping, on the Pocket page I had to work to extract full sized hero images. Exporting an image directly from the mockup clips it to the artboard, but in reality the image needs to be larger so it can overflow in a fluid layout. Hero images need to be larger than what's visible in Sketch because Sketch is not the web.

**Comment by alexgibson** [comment]
Thanks for filing this. There are a few different things I think we could do to improve this, some of which goes a bit further than documentation:

- We already list either minimum or recommended image widths for all card (and other) components in the documentation, but the component notes are a bit easy to gloss over. Could we make them more prominent?

- The values are also hard-coded, so for example when we changed token margin/padding values in https://github.com/mozilla/protocol/pull/145, all the real image dimensions changed by a few pixels. To avoid this happening in future, I wonder if we could write some JS that calculates the minimum image width for component images dynamically?

- As Craig noted, the Sketch mockups currently have a max-content width of `1024px`, which is not what is in our tokens. I spoken to @dzingeek and @michaelmccombie about this last week, and they are going to work on making sure all future Sketch files use the correct page widths, so hopefully this should help too.

- Even if we have good docs on protcol.mozilla.org for image sizing, I'm not sure if this is something designers will necessarily be looking at for a lot of the time, as they work mainly in the Sketch files. I'm not quite sure what the solution is here, but here's a couple of random thoughts:
  - Is there a place for image sizes in our tokens data?
  - We could start linking to components in Sketch from components on the docs site, and vice versa? (we should be doing this anyway)
  - Can sketch files contain notes with image sizing for designers to reference? How could we keep these in sync?

**Comment by stephaniehobson** [comment]
It would be a time saver to have the recommended height to go with the recommended width. Right now I get my calculator out every time I have figure it out. 3 times and counting on the about homepage. I should maybe document it somewhere...

**Comment by stephaniehobson** [comment]
This is what I used on /about

.mzp-l-card-half .mzp-c-card-medium: 3:2 600x400 (1800x800 @2x)
- I know it recommends 582 but the card stretches to 598, and at that point I just wanted to make it easy to remember.

.mzp-l-card-feature-left-half .mzp-c-card-feature-media 3:2 720x480 (1440x960 @2x)
- It fills the screen until 768 before switching to fill half the screen

**Comment by craigcook** [comment]
+1 on including heights for the different aspect ratios. I'm terrible at math so unless it's a really standard size I can just memorize I never know how to figure out the size from the ratio. For many years my ridiculous hack has been to make an image in Photoshop at the exact ratio I want (like 1600px b 900px for 16:9), then scale the width to my desired size (say, 598px) and see what height I end up with. I still do that sometimes if already have Photoshop open but usually I use calculateaspectratio.com. If I had to do it in my head or even with a calculator, no way.

**Comment by alexgibson** [comment]
More info and less math is always good. When I wrote my earlier reply, I was assuming the sketch component images would all be one of the three aspect ratios we support, so we wouldn’t have to worry about calculating height. But I guess we can’t always guarantee this :/

If we’re working from the Sketch library, i’d love it if we could bake in some rules to help enforce this a bit better, so we didn’t have to look these values up all the time. But adding this info to the docs now would be a good short term band-aid 👍

**Comment by craigcook** [comment]
I think it's more for having clear specs for how assets should be delivered. The Sketch library should certainly have standard sizes baked in but I'm not sure it does. It definitely doesn't have sizes large enough because it only has a 1024 desktop size. We probably also can't always count on extracting final images from a Sketch file anyway since they might be placeholders or can change up to the last minute. But if we spec out all our images sizes we can just say "we need this image at 600 x 400 and 1200 x 800" and get exactly what we need.

**Comment by michaelmccombie** [comment]
The 1024px issue in Sketch is something that we need to and will resolve soon. Unfortunately, it's not a quick fix but we are planning to do something about it. On a separate note when Vince returns we were planning on doing a design audit of all the components we have in Sketch to reduce the number of discrepancies and identify any component or token level changes that we would like to make.

It also feels a bit unfair the burden of exporting and finding assets is placed on developers alone. This is something that the designers on our team (myself included) should take more responsibility and ownership over. I'll make a couple of changes here

1) We wrote documentation on our Sketch set up. But we haven't kept it up to date and we haven't made it easy to find. I will move that documentation directly into Sketch (this is something we've wanted to do for a while) and expand upon what we've already written.
As far as images go at a minimum this will cover export settings, aspect ratios, retina images, SVG image troubleshooting, and image sizes for common components.
![image](https://user-images.githubusercontent.com/5924575/43943584-ed1f5d16-9c30-11e8-9eac-556741547e5e.png)


2) Before a Sketch file is handed over to a dev. We need to make sure these Abstract branches are consolidated, merged, and reviewed by another website designer (myself, Jennifer, Mark, and Vince). This will improve the quality of design across our site and is a chance for us to identify what new components if any will be needed. This should also be the part of the process where we are exporting the assets needed, creating new Github issues for new components, and updating the Github issue for that project. There will be instances where the assets we hand over aren't perfect so you should always feel free to jump in the Sketch file and export things yourself. But this should help.

**Comment by alexgibson** [comment]
Thanks @michaelmccombie, I think this will be really useful (in addition to improving specs in the component docs as Craig/Stephanie suggest).

**Comment by ehot** [comment]
Split component - guidelines / what kind of media query (example)

**Comment by craigcook** [comment]
This should be updated and revised now that we're doing proper responsive images. We have recommended sizes documented for some specific components but it would be good to have a general doc page for image sizing overall.
